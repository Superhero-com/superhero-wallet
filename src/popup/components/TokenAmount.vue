<template>
  <span
    class="token-amount"
    :class="{ large, 'has-label': !!label }"
  >
    <span>
      <span
        v-if="label"
        class="label"
      >
        {{ label }}
      </span>

      <span
        class="amount"
        :style="dynamicSizing
          ? { '--font-size': calculateFontSize(amountRounded) }
          : {}"
      >
        {{ amountRounded }}
        <span
          v-if="!noSymbol"
          class="symbol"
        >
          {{ symbol }}
        </span>
      </span>
    </span>
    <span
      v-if="amountFiat"
      class="fiat"
      :class="{ 'fiat-below': fiatBelow }"
    >
      {{ amountFiat }}
    </span>
  </span>
</template>

<script lang="ts">
import { computed, defineComponent } from '@vue/composition-api';
import { useCurrencies } from '../../composables';
import {
  AETERNITY_SYMBOL,
  calculateFontSize,
} from '../utils';

export default defineComponent({
  props: {
    amount: { type: Number, required: true },
    label: { type: String, default: null },
    symbol: { type: String, default: AETERNITY_SYMBOL },
    aex9: { type: Boolean, default: false },
    fiatBelow: { type: Boolean, default: false },
    hideFiat: Boolean,
    large: Boolean,
    row: Boolean,
    noSymbol: Boolean,
    highPrecision: Boolean,
    dynamicSizing: Boolean,
  },
  setup(props) {
    const { getFormattedAndRoundedFiat } = useCurrencies();

    const amountRounded = computed(() => {
      if (Number.isInteger(props.amount) || props.amount === 0) {
        return props.amount;
      }
      return props.amount.toFixed((props.highPrecision || props.amount < 0.01) ? 9 : 2);
    });

    const amountFiat = computed(
      (): string => (props.hideFiat || props.aex9) ? '' : getFormattedAndRoundedFiat(props.amount),
    );

    return {
      amountRounded,
      amountFiat,
      calculateFontSize,
    };
  },
});
</script>

<style lang="scss" scoped>
@use '../../styles/variables';
@use '../../styles/typography';

.token-amount {
  @extend %face-sans-15-medium;

  color: variables.$color-white;

  .label {
    @extend %face-sans-14-medium;

    color: rgba(variables.$color-white, 0.5);
    display: block;
  }

  .amount {
    font-size: var(--font-size);
    line-height: 20px;
  }

  .symbol {
    color: rgba(variables.$color-white, 0.75);
  }

  .fiat {
    @extend %face-sans-15-regular;

    margin-left: 8px;
    color: rgba(variables.$color-white, 0.75);

    &.fiat-below {
      display: block;
      margin-left: 0;
      padding-top: 4px;
    }
  }

  &.has-label {
    display: inline-flex;
    align-items: flex-start;
    justify-content: space-between;
    width: 100%;
  }

  &.large {
    @extend %face-sans-20-regular;

    .symbol {
      font: inherit;
    }

    .text {
      @extend %face-sans-16-regular;
    }

    .fiat {
      @extend %face-sans-18-regular;
    }
  }
}
</style>
