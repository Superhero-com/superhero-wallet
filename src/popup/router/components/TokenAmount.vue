<template>
  <span
    class="token-amount"
    :class="[direction, { large }]"
  >
    {{ amountRounded }}
    <span
      v-if="!noSymbol"
      class="symbol"
    >
      {{ symbol }}
    </span>
    <span
      v-if="amountFiat"
      class="fiat"
    >
      {{ amountFiat }}
    </span>
  </span>
</template>

<script>
import { mapState } from 'vuex';

export default {
  props: {
    amount: { type: Number, required: true },
    symbol: { type: String, default: 'AE' },
    aex9: { type: Boolean, default: false },
    hideFiat: { type: Boolean },
    direction: {
      type: String,
      validator: (value) => ['sent', 'received'].includes(value),
      default: undefined,
    },
    large: { type: Boolean },
    noSymbol: { type: Boolean },
    highPrecision: { type: Boolean },
  },
  computed: {
    amountRounded() {
      return +this.amount.toFixed((this.highPrecision || this.amount < 0.01) ? 9 : 2);
    },
    ...mapState({
      amountFiat(state, { convertToCurrency, formatCurrency }) {
        if (this.hideFiat || this.aex9) return '';
        const converted = convertToCurrency(this.amount);
        if (converted === 0) return `(${formatCurrency(0)})`;
        if (converted < 0.01) return `(<${formatCurrency(0.01)})`;
        return `(â‰ˆ${formatCurrency(converted)})`;
      },
    }),
  },
};
</script>

<style lang="scss" scoped>
@use '../../../styles/variables';
@use '../../../styles/typography';

.token-amount {
  @extend %face-sans-14-regular;

  color: variables.$color-white;

  .symbol {
    @extend %face-sans-14-medium;

    color: variables.$color-blue;
  }

  .fiat {
    color: variables.$color-dark-grey;
  }

  &.sent {
    color: variables.$color-error;

    &::before {
      content: '-';
    }
  }

  &.received {
    color: variables.$color-green-hover;

    &::before {
      content: '+';
    }
  }

  &.large {
    @extend %face-sans-20-regular;

    .symbol {
      font: inherit;
    }

    .text {
      @extend %face-sans-16-regular;
    }
  }
}
</style>
