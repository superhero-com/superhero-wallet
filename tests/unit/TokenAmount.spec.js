import { shallowMount } from '@vue/test-utils';
import TokenAmount from '../../src/popup/components/TokenAmount.vue';

let wrapper;

describe('TokenAmount', () => {
  beforeEach(() => {
    wrapper = shallowMount(TokenAmount, {
      propsData: {
        amount: 111,
        hideFiat: true,
        direction: 'sent',
        large: false,
        noSymbol: true,
        highPrecision: false,
      },
      mocks: {
        $store: {
          state: {},
          getters: {},
        },
      },
    });
  });

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
