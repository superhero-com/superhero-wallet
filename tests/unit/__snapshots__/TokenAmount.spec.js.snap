// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TokenAmount should match the snapshot 1`] = `
<span class="token-amount sent"><span><!----> <span class="amount">
      111
      <!----></span></span>
<!----></span>
`;
